sdmetrics/reports/multi_table/_properties/base.py

```diff
@@ -11,13 +11,55 @@ class BaseMultiTableProperty():
     Attributes:
         properties (dict):
             A dict mapping the table names to their single table properties.
+        is_computed (bool):
+            Whether or not the property has been computed.
+        _only_multi_table (bool):
+            Whether or not the property only exist for multi-tables.
+        details_property (pandas.DataFrame):
+            The multi table details property dataframe.
     """
 
     _single_table_property = None
 
     def __init__(self):
         self._properties = {}
         self.is_computed = False
+        self._only_multi_table = False
+        self.details_property = pd.DataFrame()
+
+    def _get_num_iterations(self, metadata):
+        """Get the number of iterations for the property."""
+        raise NotImplementedError()
```

Comment:
It feels like there's a lot of repeated code by doing it this way and forcing each property to re-define separately, especially since there's really only 3 basic cases: per column, per column pair, and per table. Maybe instead we could define each on the base and then have the child properties choose between them? Or have the base class case `_get_num_iterations` on an attribute of the child to select the appropriate variation.